src/main/java/seedu/address/model/company/UniqueCompanyList.java

```diff
@@ -0,0 +1,159 @@
+package seedu.address.model.company;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.company.exceptions.CompanyNotFoundException;
+import seedu.address.model.company.exceptions.DuplicateCompanyException;
+
+/**
+ * A list of companies that enforces uniqueness between its elements and does not allow nulls.
+ * A company is considered unique as deemed by the <Code>Company.equals(Company)</Code>.
+ * Additions and deletions use this logic to determine what can be added and removed.
+ */
+public class UniqueCompanyList implements Iterable<Company> {
+    private final ObservableList<Company> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Company> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains a company.
+     *
+     * @param toCheck Company to be checked.
+     * @return true if list already contains company.
+     */
+    public boolean contains(Company toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::equals);
```

Comment:
I just realized that your `Company::equals` does not behave like `Person::equals`. Perhaps for consistency sake, and to prevent potential confusion in the future, can you implement `Company::isSameCompany` and make `Company::equals` behave similarly to `Person::equals`?

Reply:
changed